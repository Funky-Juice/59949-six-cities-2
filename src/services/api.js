import axios from 'axios';
import ActionCreator from '../store/actions';

const apiURL = `https://htmlacademy-react-2.appspot.com/six-cities`;

export const createAPI = (dispatch) => {
  const api = axios.create({
    baseURL: apiURL,
    timeout: 5000,
    withCredentials: true
  });

  const onSuccess = (res) => res;

  const onFail = (err) => {
    if (!err.response) {
      dispatch(ActionCreator.setError(err.message));
      throw err;
    }
    if (err.response.status === 401 || err.response.status === 403) {
      dispatch(ActionCreator.requireAuthorization(true));
      throw new Error(`unauthorized`);
    }
    throw err;
  };

  api.interceptors.response.use(onSuccess, onFail);

  return api;
};
