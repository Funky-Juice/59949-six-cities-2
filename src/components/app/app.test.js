import {BrowserRouter as Router} from 'react-router-dom';
import {applyMiddleware, createStore} from 'redux';
import {createAPI} from '../../services/api';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';

import offerTestObj from '../../mocks/test-offer';
import userTestObj from '../../mocks/test-user';
import App from './app';

const mockOffers = [offerTestObj];
const createNodeMock = () => document.createElement(`div`);

const reducer = () => ({
  user: userTestObj,
  offers: mockOffers,
  activeCity: mockOffers[0].city,
  activeOffers: mockOffers
});

const api = createAPI();

const store = createStore(
    reducer,
    applyMiddleware(thunk.withExtraArgument(api))
);

it(`App correctly renders`, () => {
  const options = {createNodeMock};
  const tree = renderer
    .create(<Provider store={store}>
      <Router>
        <App
          isAuthorizationRequired={false}
          errorMessage={null}
        />
      </Router>
    </Provider>, options)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
