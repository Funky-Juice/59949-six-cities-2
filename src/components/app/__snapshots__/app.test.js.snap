// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders 1`] = `
<div
  className="cities"
>
  <div
    className="cities__places-container container"
  >
    <section
      className="cities__places places"
    >
      <div
        className="cities__places-list places__list tabs__content"
      />
    </section>
  </div>
</div>
`;
